libs/src/sdk/api/playlists/PlaylistsApi.ts

```diff
@@ -12,22 +12,165 @@ import {
   PlaylistsApi as GeneratedPlaylistsApi
 } from '../generated/default'
 import {
+  createUploadPlaylistSchema,
+  PlaylistMetadata,
   RepostPlaylistRequest,
   RepostPlaylistSchema,
   SavePlaylistRequest,
   SavePlaylistSchema,
   UnrepostPlaylistSchema,
   UnsavePlaylistRequest,
-  UnsavePlaylistSchema
+  UnsavePlaylistSchema,
+  UploadPlaylistRequest
 } from './types'
+import { retry3 } from '../../utils/retry'
+import { generateMetadataCidV1 } from '../../utils/cid'
+import { TrackUploadHelper } from '../tracks/TrackUploadHelper'
+import type { TrackMetadata } from '../tracks/types'
 
 export class PlaylistsApi extends GeneratedPlaylistsApi {
+  private readonly trackUploadHelper: TrackUploadHelper
+
   constructor(
     configuration: Configuration,
+    private readonly storage: StorageService,
     private readonly entityManager: EntityManagerService,
     private readonly auth: AuthService
   ) {
     super(configuration)
+    this.trackUploadHelper = new TrackUploadHelper(configuration)
+  }
+
+  /**
+   * Upload a playlist or album
+   */
+  async uploadPlaylist(
+    requestParameters: UploadPlaylistRequest,
+    writeOptions?: WriteOptions
+  ) {
+    // Parse inputs
+    const {
+      userId,
+      trackFiles,
+      coverArtFile,
+      metadata,
+      trackMetadatas: parsedTrackMetadatas,
+      onProgress
+    } = parseRequestParameters(
```

Comment:
this is a little but complicated, could we just create the function outside this method? like `parseUploadPlaylistParams`?

Reply:
I like the thought but we do this in every write method. I would be in favor of keeping as is because otherwise we would end up with a parse function for every write method